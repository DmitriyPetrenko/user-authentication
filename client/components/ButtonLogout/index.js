// Core
import React from "react";
import { func, object } from "prop-types";

function ButtonLogout (props) {
    return (
        <button
            className="header__menu-link"
            onClick={() => {
                props.logout(() => {
                    props.history.push("/");
                });
            }}
        >
            Logout
        </button>
    );
}

export default ButtonLogout;
