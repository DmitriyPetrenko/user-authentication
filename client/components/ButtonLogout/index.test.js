// Core
import React from "react";
import renderer from "react-test-renderer";

import ButtonLogout from "./index";

test("Button rendered correct", () => {
    const component = renderer.create(<ButtonLogout />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
