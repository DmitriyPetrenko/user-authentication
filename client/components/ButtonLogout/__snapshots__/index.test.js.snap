// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered correct 1`] = `
<button
  className="header__menu-link"
  onClick={[Function]}
>
  Logout
</button>
`;
